### DISCUSSION TITLE
Hydrating deleted model (SupportLegacyModels) leads to error.

### DISCUSSION THREAD

**Discussion by ghbob** [livewire/livewire#7987] [1. Help]
Hello! 👋

We `wire:poll` a page where a model is used. If something now deletes the model in between the following polls will fail because it can't hydrate the model again, at least that's my current assumption.  
Is this expected behaviour?  
I would assume it would be null instead and I'd need to handle the missing model in my Livewire component.

I am guessing [EloquentModelSynth.php#L207](https://github.com/livewire/livewire/blob/main/src/Features/SupportLegacyModels/EloquentModelSynth.php#L207) returns `null` and when [EloquentModelSynth.php#L89](https://github.com/livewire/livewire/blob/main/src/Features/SupportLegacyModels/EloquentModelSynth.php#L89) calls `setDataOnModel` with `null` the error occurs.

```
Livewire\Features\SupportLegacyModels\EloquentModelSynth::setDataOnModel(): Argument #1 ($model) must be of type Illuminate\Database\Eloquent\Model, null given, called in .../vendor/livewire/livewire/src/Features/SupportLegacyModels/EloquentModelSynth.php on line 80
```

Seems related to https://github.com/livewire/livewire/discussions/5979 maybe the extra null check suggested there would be a solution?

Thanks!

**Comment by joshhanley** [comment]
@ghbob yep makes sense to me. PRs are welcome 🙂

**Reply by ghbob** [reply]
Okay, I tried my best: https://github.com/livewire/livewire/pull/8047
